pkg/reconciler/knative/resources/knative_route.go

```diff
@@ -22,10 +22,21 @@ import (
 	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
 	"github.com/kubeflow/kfserving/pkg/constants"
 	"github.com/kubeflow/kfserving/pkg/utils"
+	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func CreateKnativeRoute(kfsvc *v1alpha1.KFService) *knservingv1alpha1.Route {
+type RouteBuilder struct {
+	config *v1.ConfigMap
+}
+
+func NewRouteBuilder(config *v1.ConfigMap) *RouteBuilder {
```

Comment:
How do you feel about this factory pattern? Is it better than just allowing the caller to inject directly? I'm not familiar with go best practices on this.

Reply:
If you are doing this, maybe the struct should have the specific fields that will be used instead of the entire config (and initialize them from config in this fn)?

Reply:
have mapped the json config to a proper go struct now.